server/irmaserver/sessions.go

```diff
@@ -16,46 +21,50 @@ import (
 )
 
 type session struct {
-	sync.Mutex
-	locked bool
+	//TODO: check if we can get rid of this Mutex for Redis
+	sync.Mutex `json:-`
+	//TODO: note somewhere that state with redis will not support sse for the moment
+	sse         *sse.Server
+	locked      bool
+	sessions    sessionStore
+	conf        *server.Configuration
```

Comment:
The sessionStore and the server configuration is the same for all sessions right? Can we not combine this somewhere in the `Server` struct instead of storing them for each session too?

Reply:
True, I understand the background of this, but that was not the point. Doing it in this way with Redis means that you have to initialise the configuration again always. You cannot simply do a Unmarshal to get the session again, because you always have to initialise those server-specific variables again, while they aren't related to the session at all but to the server configuration.

But I don't know whether it is easy to realize, it was just an idea I had.